phoenix-core-client/src/main/java/org/apache/phoenix/util/ViewUtil.java

```diff
@@ -886,4 +893,42 @@ public static void addTagsToPutsForViewAlteredProperties(List<Mutation> tableMet
 
         }
     }
+
+    /**
+     * Retrieves the list of unique view index ids from SYSTEM.CATALOG for a given view index table
+     * @param connection - Phoenix Connection
+     * @param tableName - name of the view index table for which view index ids need to be retrieved
+     * @param includeTenantViewIndexes - true if tenant view indexes to be included otherwise false
+     * @return - list of view index ids
+     * @throws IOException
+     * @throws SQLException
+     */
+    public static List<String> getViewIndexIds(final PhoenixConnection connection, final String tableName, final boolean includeTenantViewIndexes)
+            throws IOException, SQLException {
+        Preconditions.checkArgument(MetaDataUtil.isViewIndex(tableName));
+        List<String> viewIndexIdsString = new ArrayList<>();
+        PreparedStatement preparedStatement = connection.prepareStatement(getViewIndexIdsQuery(tableName, includeTenantViewIndexes));
```

Comment:
Good to have INFO log with the value of `getViewIndexIdsQuery()` to help debug if something goes wrong.